tinygrad/codegen/lowerer.py

```diff
@@ -58,8 +70,15 @@ def lower(self, ast:UOp, opts:Renderer) -> UOp:
         self.idxs = get_grouped_dims("idx", full_shape[:global_dims], opts.global_max, reverse=True)
       else:
         # define indexes for GPU-like execution
-        self.idxs = get_grouped_dims("gidx", full_shape[:global_dims], opts.global_max, reverse=True) + \
-                    get_grouped_dims("lidx", full_shape[global_dims:first_reduce+group_for_reduces], opts.local_max)
+        reduceops = [x for x in ast.parents if x.op is UOps.REDUCE_AXIS and all(i < first_reduce+group_for_reduces for i in x.arg[1])]
+        if len(reduceops) > 0:
```

Comment:
why is this counting local reduces?

Reply:
this is to determine whether we try to merge the `lidx`s used for local reduces, so it should only apply if there are local reduces and it shouldn't apply to reduces that before `ridx`s

Reply:
even if there's only 1 local reduceop?
ref: test_arange_big

Reply:
I figured `len(reduceops) > 1` is tighter but `len(reduceops) > 0` is simpler (are there local reducops or not?)
both are correct and I'm happy to change it to `len(reduceops) > 1`